### ISSUE TITLE
TypeError because self.readlines() is None on EOF

### ISSUE THREAD

**Issue by The-Compiler** [t-wissmann/barpyrus#14]
I just got this when doing `hc wmexec`:

```pytb
Traceback (most recent call last):
  File "/usr/bin/barpyrus", line 33, in <module>
    sys.exit(load_entry_point('barpyrus==0.0.0', 'console_scripts', 'barpyrus')())
  File "/usr/lib/python3.8/site-packages/barpyrus/mainloop.py", line 47, in main
    main_loop(bar)
  File "/usr/lib/python3.8/site-packages/barpyrus/mainloop.py", line 95, in main_loop
    x.process()
  File "/usr/lib/python3.8/site-packages/barpyrus/core.py", line 50, in process
    for line in self.readlines():
TypeError: 'NoneType' object is not iterable
```

This seems to happen because:

https://github.com/t-wissmann/barpyrus/blob/73e094e4adbeebe9b800d9e866eab76743a32dc3/barpyrus/core.py#L38-L42

but the `None` return value isn't actually handled:

https://github.com/t-wissmann/barpyrus/blob/73e094e4adbeebe9b800d9e866eab76743a32dc3/barpyrus/core.py#L49-L51

Not sure what the proper fix would be though - should it `return []` instead?

**Comment by t-wissmann** [comment]
Probably exiting is the best behaviour (the same traceback should also appear when quitting hlwm)

**Comment by t-wissmann** [comment]
Fixed by #15
